packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -31,10 +35,23 @@ export default function SidebarNavigationScreenPages() {
 		'page'
 	);
 
+	const { setIsCreatePageModalOpened } = useDispatch( editSiteStore );
+
 	return (
 		<SidebarNavigationScreen
 			title={ __( 'Pages' ) }
 			description={ __( 'Browse and edit pages on your site.' ) }
+			actions={
+				<SidebarButton
+					icon={ plus }
+					label={ __( 'Draft a new page' ) }
+					onClick={ () =>
+						setIsCreatePageModalOpened( true, {
```

Comment:
Why is this using a store value, instead of just rendering the the `AddNewPageModal` here and using local state only render `AddNewPageModal` when it's open?

Reply:
I can change to do that if you think it's a better approach. Was thinking about other triggers like Link UI or command center.

Reply:
LinkUI is in the block editor package and command center is in its own package too. I'm not sure how exactly they can share the same UI. It might be possible but there's a lot of challenges there. So I think to start, it's better to avoid any new API.

Reply:
Sounds good. I'll update tomorrow.

Reply:
Updated to use local state and onSave.

Reply:
> Was thinking about other triggers like Link UI or command center.

This is correct. And also `Navigation` section in Browse Mode.

> LinkUI is in the block editor package and command center is in its own package too. I'm not sure how exactly they can share the same UI.

The link control doesn't need to care about _how_ the create page happens. All it needs is a function it should trigger when someone clicks the appropriate piece of UI. That would be possible to pass to the `LinkControl` component via block editor settings in the same way we have had to do for many aspects of this component,

The other use case of the `Navigation` section of Browse Mode is even more clear cut as it sits adjacent to the `Pages` panel.

@youknowriad would you be open to be reverting this back to a global implementation in order that it can be triggered from elsewhere in the interface. I think "Create new page" is a pretty standard interaction for the editor as a whole.

Reply:
> The link control doesn't need to care about how the create page happens. 

Actually one thing I realised is that the Link Control would need to have a way to get the data about the created page. That could be slightly tricky as it doesn't have direct access to Core Data.

@youknowriad The inability for key editor components to access Core Data has been a long standing issue. Is it not possible to provide access to these via some kind of proxy that we can pass into block editor settings? That way the block editor package can remain de-coupled but still have a means to access the kind of data it requires.

Reply:
> The inability for key editor components to access Core Data has been a long standing issue. Is it not possible to provide access to these via some kind of proxy that we can pass into block editor settings?

It's only an issue if you think from a WordPress editor perspective. IMO, it's actually a good thing that force us to think about the right abstractions in the right place. Without these kind of constraints and abstractions, Gutenberg would not be usable in Tumblr, Drupal...